### PR TITLE
Add guides on how to use relationship fields

### PR DESCRIPTION
**Pull request by timleslie**
Add docs on the core concepts of relationships, how to decide on relationship configuration, and API docs for the relationship field itself.

### CODE REVIEW

File: packages-next/fields/src/types/relationship/index.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export type RelationshipFieldConfig<
     hideCreate?: boolean;
   };
   defaultValue?: FieldDefaultValue<Record<string, unknown>>;
+  // FIXME: NOT IMPLEMENTED
```
Comment (timleslie) [inline packages-next/fields/src/types/relationship/index.ts:47]: This feature isn't actually implemented, so I've not included it in the API docs. I'm going to follow up on what to do with the types themselves in the context of the upcoming field changes separate to this PR.

File: docs-next/pages/apis/fields.mdx [hunk 2]
```diff
@@ -424,9 +442,21 @@ export default config({
...
           ref: '...',
           many: false,
-          ui: { /* ... */ },
           defaultValue: '...',
-          isUnique: '...',
+          ui: {
+            hideCreate: false,
+            // Display mode: 'select'
+            displayMode: 'select',
+            labelField: 'name,
```
Comment (dominikwilkowski) [inline docs-next/pages/apis/fields.mdx:450]: missing a quote
```suggestion
            labelField: 'name',
```

File: docs-next/pages/guides/configuring-relationships.mdx [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    User: list({
+      fields: {
+        name: text()
+      }
+    }),
+    Post: list({
+      fields: {
+        title: text(),
+        content: text(),
+        author: relationship({ ref: 'User', many: false }),
```
Comment (dominikwilkowski) [inline docs-next/pages/guides/configuring-relationships.mdx:128]: This one has a trailing comma and the other bits don't ... not a blocker... just pointing out

Comment (timleslie) [reply docs-next/pages/guides/configuring-relationships.mdx:128]: Ah yeah, I need to go and manually lint this 👍

File: docs-next/pages/guides/configuring-relationships.mdx [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+```typescript
+import { config, createSchema, list } from '@keystone-next/keystone/schema';
+import { text, relationship } from '@keystone-next/fields';
+
+export default config({
+  lists: createSchema({
+    User: list({
+      fields: {
+        name: text()
```
Comment (dominikwilkowski) [inline docs-next/pages/guides/configuring-relationships.mdx:121]: This one doesn't have a trailing comma while the other bits do... not a blocker... just pointing out.

File: docs-next/pages/guides/configuring-relationships.mdx [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+## One-sided vs two-sided
+
+A relationship in Keystone exists _between_ two lists.
+In our blog, the concept of _authorship_ (who wrote a post) can be represented as a relationship between the `User` and the `Post` lists.
+
+The first question you need to consider is which list do you want to be able to access the relationship from in your GraphQL API?
+In our blog we might want to be able to ask about a `user's posts`, a `post's author`, or possibly both.
+If you only need to access one side of the relationship then you want to configure a _one-sided_ relationship. If you need both, then you want to configure a _two-sided_ relationship.
+
```
Comment (molomby) [inline docs-next/pages/guides/configuring-relationships.mdx:19]:
```suggestion
Note that, in either case, the data stored is the same.
This distinction solely effects how the relationship is surfaces by the GraphQL API.

```

(This is correct, right?)

File: docs-next/pages/guides/understanding-relationships.mdx [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+        title: text(),
+        content: text(),
+        authors: relationship({ ref: 'User', many: true }),
+      }
+    })
+  }),
+});
+```
+
+### Twos-sided
```
Comment (JedWatson) [inline docs-next/pages/guides/understanding-relationships.mdx:221]:
```suggestion
### Two-sided
```

**Review by JedWatson — CHANGES_REQUESTED** [review]

File: docs-next/pages/guides/configuring-relationships.mdx [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+});
+```
+
+Note that we have used `many: false` in both the `authors` and `posts` fields.
+
+## Summary
+
+When configuring a relationship in Keystone there are two key questions you need to answer:
+
+- Do I want a one-sided or two-sided relationship?
```
Comment (molomby) [inline docs-next/pages/guides/configuring-relationships.mdx:238]:
```suggestion
- Do I want to access the relationship from only one-side or both?
```

Or something similar; just feel we should be emphasising this is purely around access/API.

**Review by molomby — APPROVED** [review]
Generally looks great.

I do think we could do a better job of calling out that the one/two-sided distinction only effects the API. It's not about repeating ourselves, just about phrasing. See inline suggestions.

As mentioned, I thought some of your link paths were incorrect but apparently it's the [style guide](https://github.com/keystonejs/keystone/blob/master/STYLE_GUIDE.md#links) that needs fixing. skinner-could-it-be.jpg

Also, not one for now, but being all the "many: true | false" and one/two sided stuff explained really reinforces to me that our method of configuring this stuff is bad. It would be so much more natural to talk in terms of "one-to-many", "many-to-many", etc. rather than having to decode those categories to multiple bits of config that exist in different places. Still hope to resolve this one day.

File: docs-next/pages/guides/configuring-relationships.mdx (hunk 7 above)
Comment (timleslie) [reply docs-next/pages/guides/configuring-relationships.mdx:238]: This language is specifically goal oriented, not technically oriented, around what they want to be able to do, which will lead them to the technical decision of one-sided or two-sided.

Comment (timleslie) [reply docs-next/pages/guides/configuring-relationships.mdx:238]: Oh wait, I'm reading this backwards, and it's my prose that's very wrong. yes, good suggestion indeed!

**Pull request closed without merging**